vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -0,0 +1,227 @@
+     */
+    public DateTimePicker(
+            ValueChangeListener<ComponentValueChangeEvent<DateTimePicker, LocalDateTime>> listener) {
+        this();
+        addValueChangeListener(listener);
+    }
+
+    /**
+     * Convenience constructor to create a date time picker with a pre-selected date
+     * in current UI locale format and a {@link ValueChangeListener}.
+     *
+     * @param initialDateTime
+     *            the pre-selected date time in the picker
+     * @param listener
+     *            the listener to receive value change events
+     * @see #setValue(LocalDateTime)
+     * @see #addValueChangeListener(HasValue.ValueChangeListener)
+     */
+    public DateTimePicker(LocalDateTime initialDateTime,
+            ValueChangeListener<ComponentValueChangeEvent<DateTimePicker, LocalDateTime>> listener) {
+        this(initialDateTime);
+        addValueChangeListener(listener);
+    }
+
+    /**
+     * Convenience Constructor to create a date time picker with pre-selected date
+     * and locale setup.
+     *
+     * @param initialDateTime
+     *            the pre-selected date time in the picker
+     * @param locale
+     *            the locale for the date time picker
+     */
+    public DateTimePicker(LocalDateTime initialDateTime, Locale locale) {
+        this(initialDateTime);
+        setLocale(locale);
+    }
+
+    /**
+     * This method should return the value of the field, based on value of the internal fields.
+     *
+     * @return new value of the field.
+     */
+    protected LocalDateTime generateModelValue() {
+        final LocalDate date = datePicker.getValue();
+        final LocalTime time = timePicker.getValue();
+        return date != null && time != null ?
+                LocalDateTime.of(date, time) :
+                null;
+    }
+
+    @Override
+    protected void setPresentationValue(
+            LocalDateTime newPresentationValue) {
+        datePicker.setValue(newPresentationValue != null ?
+                newPresentationValue.toLocalDate() :
+                null);
+        timePicker.setValue(newPresentationValue != null ?
+                newPresentationValue.toLocalTime() :
+                null);
+    }
+
+    /**
+     * Regenerates the value by calling {@link #generateModelValue()} and updates the model.
+     * If the value is different than the current one, a {@link ValueChangeEvent} will be generated with {@link ValueChangeEvent#isFromClient()} set to <code>true</code>
+     *
+     * <p>This method is called when the web component generates a <b>changed</b> event, typically in response to a change made by the user in one of the contained fields.
+     *
+     * <p>Custom implementations of this method <b>must</b> call {@link #setModelValue(LocalDateTime, boolean)} with the updated model value.
+     * Subclasses can call this method when the model value needs to be regenerated and updated.
+     */
+    protected void updateValue() {
```

Comment:
Do you have a use case why this should be exposed as `protected`?

Reply:
This was mimicked from CustomField: https://github.com/vaadin/vaadin-custom-field-flow/blob/master/vaadin-custom-field-flow/src/main/java/com/vaadin/flow/component/customfield/CustomField.java#L104

I guess in this case it might as well be private?